app/components/swatch.js

```diff
@@ -0,0 +1,23 @@
+import Component from '@ember/component';
+import { computed} from '@ember/object';
+import { htmlSafe } from '@ember/string';
+
+export default class Swatch extends Component{
+    @computed('shade','size')  
+    get style() {
+        const size = this.translateSize(this.size);
+        return htmlSafe(`background-color: ${this.shade}; width:${size}px; height:${size}px`);
+    }
+
+
+    translateSize(size="s") {
+        switch(size){
+            case("s"):
+                return 10;
+            case("m"):
+                return 20;
+            case("l"):
+                return 30;
+        }
+    }
```

Comment:
Neat! — I like the ability to add sizes in here like this.

Reply:
If the only logic here is to return a numeric size, would it be simpler to have an object that maps out each size to its numeric counterpart? The object could live outside of the `Swatch` class, and then in side the class you could do something like:

`const size = availableSizes[size] || 10`